src/sst/elements/ariel/arielmemmgr.h

```diff
@@ -34,12 +39,41 @@ class ArielMemoryManager : public SubComponent {
     public:
         SST_ELI_REGISTER_SUBCOMPONENT_API(SST::ArielComponent::ArielMemoryManager)
 
+        SST_ELI_DOCUMENT_PARAMS(
```

Comment:
Fundamentally, is there a reason you can't create an OpalMemoryManager subcomponent for Ariel that does all these things (and inherits from ArielMemoryManager) instead of modifying the base class for all Ariel memory managers? Along the lines of how the other ArielMemoryManager subcomponents work? Is there something that needs to be in the base ArielMemoryManager API that does not exist right now?

Reply:
@gvoskuilen Ariel memory manager has to communicate with the Opal to send hints during malloc and mmap calls. This just requires two links to be added to the ariel memory manager. These links will only be activated if the Opal component is enabled. I thought of creating generic links from ariel memory manager to Opal which does not depend on opal_event (later on) instead of adding two subcomponents to the ariel component (opal_meorymanager and a translator).

But for now, as you suggested I have removed all the opal related code from ariel and updated the opal test configuration. I am also working on doing the same for the Samba component.